### ISSUE TITLE
Pods failing to remount storage on restart.

### ISSUE THREAD

**Issue by owenmonagan** [arangodb/kube-arangodb#771]
My cluster always seems to get into a broken state.

I suspect its because I'm running on GKE with preemptible VMs as my nodes. The nodes are frequently replaced, and so are the pods running on the nodes. But I would also swear that I had this problem before using non preemptible nodes with GKE.

> A preemptible VM is an instance that you can create and run at a much lower price than normal instances. However, Compute Engine might stop (preempt) these instances if it requires access to those resources for other tasks. Preemptible instances are excess Compute Engine capacity, so their availability varies with usage.

How can I fine tune my cluster to be fault tolerant in case of a node being replaced?

```
arango-kube-arangodb-operator-658b864c8-hjp5t   1/1     Running    0          19m
arango-kube-arangodb-operator-658b864c8-nwdcd   1/1     Running    0          44m
arangodb-agnt-0l8projf-d27445                   0/2     Init:0/2   0          169m
arangodb-agnt-i4szb8we-d27445                   0/2     Init:0/2   0          15h
arangodb-agnt-zv6x9rmo-d27445                   0/2     Init:0/2   0          59m
arangodb-crdn-jstbdklm-d27445                   1/2     Running    0          96s
arangodb-crdn-nuagvixf-d27445                   1/2     Running    0          96s
arangodb-crdn-xhdsl23p-d27445                   1/2     Running    0          96s
arangodb-prmr-jvyspyz8-d27445                   0/2     Init:0/2   0          59m
arangodb-prmr-ypyxx999-d27445                   0/2     Init:0/2   0          161m
arangodb-prmr-zhlbdec9-d27445                   0/2     Init:0/2   0          42m
arangodb-storage-b6gn6                          1/1     Running    0          55m
arangodb-storage-bzwss                          1/1     Running    0          15h
arangodb-storage-l8c9c                          1/1     Running    0          14m
arangodb-storage-n227q                          1/1     Running    0          168m
arangodb-storage-p6lpt                          1/1     Running    0          44m
```

```
kc describe pod arangodb-agnt-0l8projf-d27445                                                                                                                                          ✔
Name:           arangodb-agnt-0l8projf-d27445
Namespace:      arangodb
Priority:       0
Node:           gke-cluster-01-group-01-ea59db84-vhtj/10.128.15.194
Start Time:     Tue, 03 Aug 2021 17:32:08 -0400
Labels:         app=arangodb
                arango_deployment=arangodb
                arango_exporter=yes
                deployment.arangodb.com/member=AGNT-0l8projf
                role=agent
Annotations:    <none>
Status:         Pending
IP:
IPs:            <none>
Controlled By:  ArangoDeployment/arangodb
Init Containers:
  init-lifecycle:
    Container ID:
    Image:         arangodb/kube-arangodb:1.2.1
    Image ID:
    Port:          <none>
    Host Port:     <none>
    Command:
      /usr/bin/arangodb_operator
      lifecycle
      copy
      --target
      /lifecycle/tools
    State:          Waiting
      Reason:       PodInitializing
    Ready:          False
    Restart Count:  0
    Environment:    <none>
    Mounts:
      /lifecycle/tools from lifecycle (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from default-token-tvg9h (ro)
  uuid:
    Container ID:
    Image:         arangodb/kube-arangodb:1.2.1
    Image ID:
    Port:          <none>
    Host Port:     <none>
    Command:
      /usr/bin/arangodb_operator
      uuid
      --uuid-path
      /data/UUID
      --engine-path
      /data/ENGINE
      --uuid
      AGNT-0l8projf
      --engine
      rocksdb
    State:          Waiting
      Reason:       PodInitializing
    Ready:          False
    Restart Count:  0
    Limits:
      cpu:     100m
      memory:  50Mi
    Requests:
      cpu:        100m
      memory:     10Mi
    Environment:  <none>
    Mounts:
      /data from arangod-data (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from default-token-tvg9h (ro)
Containers:
  server:
    Container ID:
    Image:         docker.io/arangodb/arangodb@sha256:f8998c09106e0dea2a02da26f306ad50989a31d5083ca1c71afe10cc249950de
    Image ID:
    Port:          8529/TCP
    Host Port:     0/TCP
    Command:
      /usr/sbin/arangod
      --agency.activate=true
      --agency.disaster-recovery-id=AGNT-0l8projf
      --agency.endpoint=tcp://arangodb-agent-i4szb8we.arangodb-int.arangodb.svc:8529
      --agency.endpoint=tcp://arangodb-agent-zv6x9rmo.arangodb-int.arangodb.svc:8529
      --agency.my-address=tcp://arangodb-agent-0l8projf.arangodb-int.arangodb.svc:8529
      --agency.size=3
      --agency.supervision=true
      --database.directory=/data
      --foxx.queues=false
      --log.level=INFO
      --log.output=+
      --server.authentication=true
      --server.endpoint=tcp://[::]:8529
      --server.jwt-secret-keyfile=/secrets/cluster/jwt/token
      --server.statistics=false
      --server.storage-engine=rocksdb
    State:          Waiting
      Reason:       PodInitializing
    Ready:          False
    Restart Count:  0
    Liveness:       exec [/lifecycle/tools/arangodb_operator lifecycle probe --endpoint=/_api/version --auth] delay=900s timeout=2s period=60s #success=1 #failure=10
    Environment:
      MY_POD_NAME:       arangodb-agnt-0l8projf-d27445 (v1:metadata.name)
      MY_POD_NAMESPACE:  arangodb (v1:metadata.namespace)
      MY_NODE_NAME:       (v1:spec.nodeName)
      NODE_NAME:          (v1:spec.nodeName)
    Mounts:
      /data from arangod-data (rw)
      /lifecycle/tools from lifecycle (rw)
      /secrets/cluster/jwt from cluster-jwt (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from default-token-tvg9h (ro)
  exporter:
    Container ID:
    Image:         docker.io/arangodb/arangodb@sha256:f8998c09106e0dea2a02da26f306ad50989a31d5083ca1c71afe10cc249950de
    Image ID:
    Port:          9101/TCP
    Host Port:     0/TCP
    Command:
      /lifecycle/tools/arangodb_operator
      exporter
      --arangodb.endpoint=http://localhost:8529/_admin/metrics
      --arangodb.jwt-file=/secrets/exporter/jwt/token
    State:          Waiting
      Reason:       PodInitializing
    Ready:          False
    Restart Count:  0
    Liveness:       http-get http://:9101/ delay=900s timeout=2s period=60s #success=1 #failure=10
    Environment:    <none>
    Mounts:
      /lifecycle/tools from lifecycle (rw)
      /secrets/exporter/jwt from exporter-jwt (ro)
      /var/run/secrets/kubernetes.io/serviceaccount from default-token-tvg9h (ro)
Conditions:
  Type              Status
  Initialized       False
  Ready             False
  ContainersReady   False
  PodScheduled      True
Volumes:
  arangod-data:
    Type:       PersistentVolumeClaim (a reference to a PersistentVolumeClaim in the same namespace)
    ClaimName:  arangodb-agent-0l8projf
    ReadOnly:   false
  exporter-jwt:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  arangodb-exporter-jwt-token
    Optional:    false
  cluster-jwt:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  arangodb-jwt
    Optional:    false
  lifecycle:
    Type:       EmptyDir (a temporary directory that shares a pod's lifetime)
    Medium:
    SizeLimit:  <unset>
  default-token-tvg9h:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  default-token-tvg9h
    Optional:    false
QoS Class:       Burstable
Node-Selectors:  <none>
Tolerations:     node.alpha.kubernetes.io/unreachable:NoExecute
                 node.kubernetes.io/not-ready:NoExecute
                 node.kubernetes.io/unreachable:NoExecute
Events:
  Type     Reason       Age                    From                                            Message
  ----     ------       ----                   ----                                            -------
  Warning  FailedMount  55m (x11 over 159m)    kubelet, gke-cluster-01-group-01-ea59db84-vhtj  Unable to attach or mount volumes: unmounted volumes=[arangod-data], unattached volumes=[cluster-jwt exporter-jwt lifecycle default-token-tvg9h arangod-data]: timed out waiting for the condition
  Warning  FailedMount  30m (x7 over 164m)     kubelet, gke-cluster-01-group-01-ea59db84-vhtj  Unable to attach or mount volumes: unmounted volumes=[arangod-data], unattached volumes=[default-token-tvg9h arangod-data cluster-jwt exporter-jwt lifecycle]: timed out waiting for the condition
  Warning  FailedMount  20m (x82 over 170m)    kubelet, gke-cluster-01-group-01-ea59db84-vhtj  MountVolume.NewMounter initialization failed for volume "arangodb-storage-8dd222-zivl0cuw5fgyif2q" : path "/var/lib/arango-storage/zivl0cuw5fgyif2q" does not exist
  Warning  FailedMount  10m (x21 over 168m)    kubelet, gke-cluster-01-group-01-ea59db84-vhtj  Unable to attach or mount volumes: unmounted volumes=[arangod-data], unattached volumes=[lifecycle default-token-tvg9h arangod-data cluster-jwt exporter-jwt]: timed out waiting for the condition
  Warning  FailedMount  5m49s (x16 over 166m)  kubelet, gke-cluster-01-group-01-ea59db84-vhtj  Unable to attach or mount volumes: unmounted volumes=[arangod-data], unattached volumes=[exporter-jwt lifecycle default-token-tvg9h arangod-data cluster-jwt]: timed out waiting for the condition
```

```
kc describe pod arangodb-prmr-ypyxx999-d27445                                                                                                                                          ✔
Name:           arangodb-prmr-ypyxx999-d27445
Namespace:      arangodb
Priority:       0
Node:           gke-cluster-01-group-01-9f9e0a5b-grcd/10.128.15.197
Start Time:     Tue, 03 Aug 2021 17:42:46 -0400
Labels:         app=arangodb
                arango_deployment=arangodb
                arango_exporter=yes
                deployment.arangodb.com/member=PRMR-ypyxx999
                role=dbserver
Annotations:    <none>
Status:         Pending
IP:
IPs:            <none>
Controlled By:  ArangoDeployment/arangodb
Init Containers:
  init-lifecycle:
    Container ID:
    Image:         arangodb/kube-arangodb:1.2.1
    Image ID:
    Port:          <none>
    Host Port:     <none>
    Command:
      /usr/bin/arangodb_operator
      lifecycle
      copy
      --target
      /lifecycle/tools
    State:          Waiting
      Reason:       PodInitializing
    Ready:          False
    Restart Count:  0
    Environment:    <none>
    Mounts:
      /lifecycle/tools from lifecycle (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from default-token-tvg9h (ro)
  uuid:
    Container ID:
    Image:         arangodb/kube-arangodb:1.2.1
    Image ID:
    Port:          <none>
    Host Port:     <none>
    Command:
      /usr/bin/arangodb_operator
      uuid
      --uuid-path
      /data/UUID
      --engine-path
      /data/ENGINE
      --uuid
      PRMR-ypyxx999
      --engine
      rocksdb
      --require
    State:          Waiting
      Reason:       PodInitializing
    Ready:          False
    Restart Count:  0
    Limits:
      cpu:     100m
      memory:  50Mi
    Requests:
      cpu:        100m
      memory:     10Mi
    Environment:  <none>
    Mounts:
      /data from arangod-data (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from default-token-tvg9h (ro)
Containers:
  server:
    Container ID:
    Image:         docker.io/arangodb/arangodb@sha256:f8998c09106e0dea2a02da26f306ad50989a31d5083ca1c71afe10cc249950de
    Image ID:
    Port:          8529/TCP
    Host Port:     0/TCP
    Command:
      /usr/sbin/arangod
      --cluster.agency-endpoint=tcp://arangodb-agent-0l8projf.arangodb-int.arangodb.svc:8529
      --cluster.agency-endpoint=tcp://arangodb-agent-i4szb8we.arangodb-int.arangodb.svc:8529
      --cluster.agency-endpoint=tcp://arangodb-agent-zv6x9rmo.arangodb-int.arangodb.svc:8529
      --cluster.my-address=tcp://arangodb-dbserver-ypyxx999.arangodb-int.arangodb.svc:8529
      --cluster.my-role=PRIMARY
      --database.directory=/data
      --foxx.queues=false
      --log.level=INFO
      --log.output=+
      --server.authentication=true
      --server.endpoint=tcp://[::]:8529
      --server.jwt-secret-keyfile=/secrets/cluster/jwt/token
      --server.statistics=true
      --server.storage-engine=rocksdb
    State:          Waiting
      Reason:       PodInitializing
    Ready:          False
    Restart Count:  0
    Liveness:       exec [/lifecycle/tools/arangodb_operator lifecycle probe --endpoint=/_api/version --auth] delay=900s timeout=2s period=60s #success=1 #failure=10
    Environment:
      MY_POD_NAME:       arangodb-prmr-ypyxx999-d27445 (v1:metadata.name)
      MY_POD_NAMESPACE:  arangodb (v1:metadata.namespace)
      MY_NODE_NAME:       (v1:spec.nodeName)
      NODE_NAME:          (v1:spec.nodeName)
    Mounts:
      /data from arangod-data (rw)
      /lifecycle/tools from lifecycle (rw)
      /secrets/cluster/jwt from cluster-jwt (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from default-token-tvg9h (ro)
  exporter:
    Container ID:
    Image:         docker.io/arangodb/arangodb@sha256:f8998c09106e0dea2a02da26f306ad50989a31d5083ca1c71afe10cc249950de
    Image ID:
    Port:          9101/TCP
    Host Port:     0/TCP
    Command:
      /lifecycle/tools/arangodb_operator
      exporter
      --arangodb.endpoint=http://localhost:8529/_admin/metrics
      --arangodb.jwt-file=/secrets/exporter/jwt/token
    State:          Waiting
      Reason:       PodInitializing
    Ready:          False
    Restart Count:  0
    Liveness:       http-get http://:9101/ delay=900s timeout=2s period=60s #success=1 #failure=10
    Environment:    <none>
    Mounts:
      /lifecycle/tools from lifecycle (rw)
      /secrets/exporter/jwt from exporter-jwt (ro)
      /var/run/secrets/kubernetes.io/serviceaccount from default-token-tvg9h (ro)
Conditions:
  Type              Status
  Initialized       False
  Ready             False
  ContainersReady   False
  PodScheduled      True
Volumes:
  arangod-data:
    Type:       PersistentVolumeClaim (a reference to a PersistentVolumeClaim in the same namespace)
    ClaimName:  arangodb-dbserver-ypyxx999
    ReadOnly:   false
  exporter-jwt:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  arangodb-exporter-jwt-token
    Optional:    false
  cluster-jwt:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  arangodb-jwt
    Optional:    false
  lifecycle:
    Type:       EmptyDir (a temporary directory that shares a pod's lifetime)
    Medium:
    SizeLimit:  <unset>
  default-token-tvg9h:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  default-token-tvg9h
    Optional:    false
QoS Class:       Burstable
Node-Selectors:  <none>
Tolerations:     node.alpha.kubernetes.io/unreachable:NoExecute for 300s
                 node.kubernetes.io/not-ready:NoExecute for 300s
                 node.kubernetes.io/unreachable:NoExecute for 300s
Events:
  Type     Reason       Age                  From                                            Message
  ----     ------       ----                 ----                                            -------
  Warning  FailedMount  41m (x21 over 152m)  kubelet, gke-cluster-01-group-01-9f9e0a5b-grcd  Unable to attach or mount volumes: unmounted volumes=[arangod-data], unattached volumes=[lifecycle default-token-tvg9h arangod-data cluster-jwt exporter-jwt]: timed out waiting for the condition
  Warning  FailedMount  30m (x72 over 161m)  kubelet, gke-cluster-01-group-01-9f9e0a5b-grcd  MountVolume.NewMounter initialization failed for volume "arangodb-storage-28a5e4-io5iwsz2aamronej" : path "/var/lib/arango-storage/io5iwsz2aamronej" does not exist
  Warning  FailedMount  25m (x8 over 115m)   kubelet, gke-cluster-01-group-01-9f9e0a5b-grcd  Unable to attach or mount volumes: unmounted volumes=[arangod-data], unattached volumes=[default-token-tvg9h arangod-data cluster-jwt exporter-jwt lifecycle]: timed out waiting for the condition
  Warning  FailedMount  8m58s (x3 over 15m)  kubelet, gke-cluster-01-group-01-9f9e0a5b-grcd  Unable to attach or mount volumes: unmounted volumes=[arangod-data], unattached volumes=[default-token-tvg9h arangod-data cluster-jwt exporter-jwt lifecycle]: timed out waiting for the condition
  Warning  FailedMount  4m24s (x3 over 11m)  kubelet, gke-cluster-01-group-01-9f9e0a5b-grcd  Unable to attach or mount volumes: unmounted volumes=[arangod-data], unattached volumes=[lifecycle default-token-tvg9h arangod-data cluster-jwt exporter-jwt]: timed out waiting for the condition
  Warning  FailedMount  2m10s                kubelet, gke-cluster-01-group-01-9f9e0a5b-grcd  Unable to attach or mount volumes: unmounted volumes=[arangod-data], unattached volumes=[cluster-jwt exporter-jwt lifecycle default-token-tvg9h arangod-data]: timed out waiting for the condition
  Warning  FailedMount  87s (x16 over 17m)   kubelet, gke-cluster-01-group-01-9f9e0a5b-grcd  MountVolume.NewMounter initialization failed for volume "arangodb-storage-28a5e4-io5iwsz2aamronej" : path "/var/lib/arango-storage/io5iwsz2aamronej" does not exist
```
```
kc logs -f arangodb-storage-n227q                                                                                                                                                      ✔
2021-08-03T21:32:28Z INF Starting arangodb local storage provisioner (Community), version 1.2.1 build 01bb8b61
2021-08-03T21:32:28Z INF Listening on 0.0.0.0:8929 component=provisioner
```

```
apiVersion: "database.arangodb.com/v1alpha"
kind: "ArangoDeployment"
metadata:
  name: "arangodb"
spec:
  mode: Cluster
  environment: Production
  downtimeAllowed: true
  externalAccess:
    type: NodePort
  metrics:
    enabled: true
  tls:
    caSecretName: None
  recovery:
    autoRecover: true
  image: 'arangodb/arangodb:3.7.10'
```
```
apiVersion: "storage.arangodb.com/v1alpha"
kind: "ArangoLocalStorage"
metadata:
  name: "arangodb-storage"
spec:
  storageClass:
    name: arangodb-ssd
    isDefault: true
  localPath:
    - /var/lib/arango-storage
```
```
kind: StorageClass
apiVersion: storage.k8s.io/v1
metadata:
  name: arangodb-ssd
provisioner: kubernetes.io/no-provisioner
parameters:
  type: pd-ssd
volumeBindingMode: WaitForFirstConsumer
```

**Comment by owenmonagan** [comment]
Scaled up to a 6 node cluster. And redeployed everything last night. Woke up to the cluster breaking again.
```
 kc get pods                                                                                                                                                                                                                                ✔
NAME                                            READY   STATUS    RESTARTS   AGE
arango-kube-arangodb-operator-658b864c8-5rn2f   1/1     Running   0          78m
arango-kube-arangodb-operator-658b864c8-sprjt   1/1     Running   0          78m
arangodb-agnt-15wnupbq-d27445                   2/2     Running   0          8m20s
arangodb-agnt-cwfas4qm-d27445                   2/2     Running   0          8m19s
arangodb-agnt-dqvficp9-d27445                   2/2     Running   0          8m18s
arangodb-crdn-5xauckve-d27445                   2/2     Running   0          8m14s
arangodb-crdn-6rwx0b2v-d27445                   2/2     Running   0          8m12s
arangodb-crdn-zl9zktad-d27445                   2/2     Running   0          8m11s
arangodb-prmr-iljs2wci-d27445                   2/2     Running   0          8m17s
arangodb-prmr-jpmkrpdc-d27445                   2/2     Running   0          8m16s
arangodb-prmr-r6ftbkia-d27445                   2/2     Running   0          8m15s
arangodb-storage-2snqx                          1/1     Running   0          35m
arangodb-storage-5qt5x                          1/1     Running   0          23m
arangodb-storage-89mcs                          1/1     Running   0          32m
arangodb-storage-nbx9f                          1/1     Running   0          35m
arangodb-storage-rbdvn                          1/1     Running   0          35m
arangodb-storage-tdqsg                          1/1     Running   0          23m
arangodb-storage-xn6vx                          1/1     Running   0          23m
```
```
kc describe pod arangodb-agnt-15wnupbq-d27445                                                                                                                         1 ↵
Name:           arangodb-agnt-15wnupbq-d27445
Namespace:      arangodb
Priority:       0
Node:           gke-cluster-01-group-02-72393856-zj4w/10.128.15.214
Start Time:     Wed, 04 Aug 2021 08:20:02 -0400
Labels:         app=arangodb
                arango_deployment=arangodb
                arango_exporter=yes
                deployment.arangodb.com/member=AGNT-15wnupbq
                role=agent
Annotations:    <none>
Status:         Pending
IP:
IPs:            <none>
Controlled By:  ArangoDeployment/arangodb
Init Containers:
  init-lifecycle:
    Container ID:
    Image:         arangodb/kube-arangodb:1.2.1
    Image ID:
    Port:          <none>
    Host Port:     <none>
    Command:
      /usr/bin/arangodb_operator
      lifecycle
      copy
      --target
      /lifecycle/tools
    State:          Waiting
      Reason:       PodInitializing
    Ready:          False
    Restart Count:  0
    Environment:    <none>
    Mounts:
      /lifecycle/tools from lifecycle (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from default-token-tvg9h (ro)
  uuid:
    Container ID:
    Image:         arangodb/kube-arangodb:1.2.1
    Image ID:
    Port:          <none>
    Host Port:     <none>
    Command:
      /usr/bin/arangodb_operator
      uuid
      --uuid-path
      /data/UUID
      --engine-path
      /data/ENGINE
      --uuid
      AGNT-15wnupbq
      --engine
      rocksdb
    State:          Waiting
      Reason:       PodInitializing
    Ready:          False
    Restart Count:  0
    Limits:
      cpu:     100m
      memory:  50Mi
    Requests:
      cpu:        100m
      memory:     10Mi
    Environment:  <none>
    Mounts:
      /data from arangod-data (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from default-token-tvg9h (ro)
Containers:
  server:
    Container ID:
    Image:         docker.io/arangodb/arangodb@sha256:f8998c09106e0dea2a02da26f306ad50989a31d5083ca1c71afe10cc249950de
    Image ID:
    Port:          8529/TCP
    Host Port:     0/TCP
    Command:
      /usr/sbin/arangod
      --agency.activate=true
      --agency.disaster-recovery-id=AGNT-15wnupbq
      --agency.endpoint=tcp://arangodb-agent-cwfas4qm.arangodb-int.arangodb.svc:8529
      --agency.endpoint=tcp://arangodb-agent-dqvficp9.arangodb-int.arangodb.svc:8529
      --agency.my-address=tcp://arangodb-agent-15wnupbq.arangodb-int.arangodb.svc:8529
      --agency.size=3
      --agency.supervision=true
      --database.directory=/data
      --foxx.queues=false
      --log.level=INFO
      --log.output=+
      --server.authentication=true
      --server.endpoint=tcp://[::]:8529
      --server.jwt-secret-keyfile=/secrets/cluster/jwt/token
      --server.statistics=false
      --server.storage-engine=rocksdb
    State:          Waiting
      Reason:       PodInitializing
    Ready:          False
    Restart Count:  0
    Liveness:       exec [/lifecycle/tools/arangodb_operator lifecycle probe --endpoint=/_api/version --auth] delay=900s timeout=2s period=60s #success=1 #failure=10
    Environment:
      MY_POD_NAME:       arangodb-agnt-15wnupbq-d27445 (v1:metadata.name)
      MY_POD_NAMESPACE:  arangodb (v1:metadata.namespace)
      MY_NODE_NAME:       (v1:spec.nodeName)
      NODE_NAME:          (v1:spec.nodeName)
    Mounts:
      /data from arangod-data (rw)
      /lifecycle/tools from lifecycle (rw)
      /secrets/cluster/jwt from cluster-jwt (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from default-token-tvg9h (ro)
  exporter:
    Container ID:
    Image:         docker.io/arangodb/arangodb@sha256:f8998c09106e0dea2a02da26f306ad50989a31d5083ca1c71afe10cc249950de
    Image ID:
    Port:          9101/TCP
    Host Port:     0/TCP
    Command:
      /lifecycle/tools/arangodb_operator
      exporter
      --arangodb.endpoint=http://localhost:8529/_admin/metrics
      --arangodb.jwt-file=/secrets/exporter/jwt/token
    State:          Waiting
      Reason:       PodInitializing
    Ready:          False
    Restart Count:  0
    Liveness:       http-get http://:9101/ delay=900s timeout=2s period=60s #success=1 #failure=10
    Environment:    <none>
    Mounts:
      /lifecycle/tools from lifecycle (rw)
      /secrets/exporter/jwt from exporter-jwt (ro)
      /var/run/secrets/kubernetes.io/serviceaccount from default-token-tvg9h (ro)
Conditions:
  Type              Status
  Initialized       False
  Ready             False
  ContainersReady   False
  PodScheduled      True
Volumes:
  arangod-data:
    Type:       PersistentVolumeClaim (a reference to a PersistentVolumeClaim in the same namespace)
    ClaimName:  arangodb-agent-15wnupbq
    ReadOnly:   false
  exporter-jwt:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  arangodb-exporter-jwt-token
    Optional:    false
  cluster-jwt:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  arangodb-jwt
    Optional:    false
  lifecycle:
    Type:       EmptyDir (a temporary directory that shares a pod's lifetime)
    Medium:
    SizeLimit:  <unset>
  default-token-tvg9h:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  default-token-tvg9h
    Optional:    false
QoS Class:       Burstable
Node-Selectors:  <none>
Tolerations:     node.alpha.kubernetes.io/unreachable:NoExecute
                 node.kubernetes.io/not-ready:NoExecute
                 node.kubernetes.io/unreachable:NoExecute
Events:
  Type     Reason             Age                  From                                            Message
  ----     ------             ----                 ----                                            -------
  Normal   NotTriggerScaleUp  51m                  cluster-autoscaler                              pod didn't trigger scale-up (it wouldn't fit if a new node is added): 1 max node group size reached, 5 node(s) had volume node affinity conflict
  Warning  FailedScheduling   50m                  default-scheduler                               0/6 nodes are available: 1 Insufficient cpu, 5 node(s) had volume node affinity conflict.
  Warning  FailedScheduling   50m (x5 over 51m)    default-scheduler                               0/7 nodes are available: 1 Insufficient cpu, 1 node(s) had taint {node.kubernetes.io/not-ready: }, that the pod didn't tolerate, 5 node(s) had volume node affinity conflict.
  Normal   Scheduled          50m                  default-scheduler                               Successfully assigned arangodb/arangodb-agnt-15wnupbq-d27445 to gke-cluster-01-group-02-72393856-zj4w
  Warning  FailedMount        49m (x7 over 50m)    kubelet, gke-cluster-01-group-02-72393856-zj4w  MountVolume.NewMounter initialization failed for volume "arangodb-storage-2e4318-wlbdihtitq5ffwbh" : path "/var/lib/arango-storage/wlbdihtitq5ffwbh" does not exist
  Warning  FailedMount        31m                  kubelet, gke-cluster-01-group-02-72393856-zj4w  Unable to attach or mount volumes: unmounted volumes=[arangod-data], unattached volumes=[default-token-tvg9h arangod-data cluster-jwt exporter-jwt lifecycle]: timed out waiting for the condition
  Warning  FailedMount        24m (x4 over 36m)    kubelet, gke-cluster-01-group-02-72393856-zj4w  Unable to attach or mount volumes: unmounted volumes=[arangod-data], unattached volumes=[lifecycle default-token-tvg9h arangod-data cluster-jwt exporter-jwt]: timed out waiting for the condition
  Warning  FailedMount        22m (x3 over 40m)    kubelet, gke-cluster-01-group-02-72393856-zj4w  Unable to attach or mount volumes: unmounted volumes=[arangod-data], unattached volumes=[arangod-data cluster-jwt exporter-jwt lifecycle default-token-tvg9h]: timed out waiting for the condition
  Warning  FailedMount        12m (x23 over 42m)   kubelet, gke-cluster-01-group-02-72393856-zj4w  MountVolume.NewMounter initialization failed for volume "arangodb-storage-2e4318-wlbdihtitq5ffwbh" : path "/var/lib/arango-storage/wlbdihtitq5ffwbh" does not exist
  Warning  FailedMount        6m47s                kubelet, gke-cluster-01-group-02-72393856-zj4w  Unable to attach or mount volumes: unmounted volumes=[arangod-data], unattached volumes=[cluster-jwt exporter-jwt lifecycle default-token-tvg9h arangod-data]: timed out waiting for the condition
  Warning  FailedMount        2m18s (x4 over 38m)  kubelet, gke-cluster-01-group-02-72393856-zj4w  Unable to attach or mount volumes: unmounted volumes=[arangod-data], unattached volumes=[exporter-jwt lifecycle default-token-tvg9h arangod-data cluster-jwt]: timed out waiting for the condition
```

**Comment by ajanikow** [comment]
Hello!

In your case, when disk is not persisted, i would not suggest to use ArangoStorage extension. It works well when you are on BareMetal or on static setup. I would suggest you to use SSD storageClass (as I remember it was called Gold) provided by GKE, so your data can be just attached to new server when old one was removed.

ArangoStorage creates path only on PV create event. After that, it does not check if directory was removed. If directory disappear it means that your data was erased by external action, so it can just break your cluster. It can happen if mount was recreated, disk removed or it is mounted to ephemeral storage.

Best Regards,
Adam.

**Comment by owenmonagan** [comment]
Thanks Adam.

I managed to get it working with preemptible vm's.

I created separate google cloud disks.

My cluster has 3 agents, so I created 3 ssd's with 8gb storage.

My cluster has 3 servers, so I created 3 ssd's with 80gb storage.

I then create persistent volumes in my k8s cluster, (https://cloud.google.com/kubernetes-engine/docs/concepts/persistent-volumes). I labeled the agent disks and server disks differently.

I disabled the arango storage operator.

I then used the persistent volume claim template in the arango deployment object to reference the persistent volumes via labels.

My cluster has been healthy for about a week now.
